Framework/Kernel/inc/MantidKernel/Cache.h

```diff
@@ -74,7 +99,26 @@ template <class KEYTYPE, class VALUETYPE> class DLLExport Cache {
    * @param value The new value for the key
    */
   void setCache(const KEYTYPE &key, const VALUETYPE &value) {
-    m_cacheMap[key] = value;
+    //
+    auto where = m_cacheMap.find(key);
```

Comment:
Now thread A can run `find` while thread B modifies `m_cacheMap`. Are you sure that is possible with `tbb::concurrent_unordered_map`? It looks wrong to me.